rmw_connext_shared_cpp/src/create_topic.cpp

```diff
@@ -40,14 +40,11 @@ rmw_connext_shared_cpp::create_topic(
 
   std::lock_guard<std::mutex> guard(node_info->topic_creation_mutex);
 
-  DDS::TopicDescription * topic_description =
-    participant->lookup_topicdescription(dds_topic_name);
-
-  DDS::Topic * topic = nullptr;
-  DDS::ReturnCode_t status = DDS::RETCODE_ERROR;
-  if (!topic_description) {
+  DDS::Duration_t timeout = DDS::Duration_t::from_seconds(0);
+  DDS::Topic * topic = participant->find_topic(dds_topic_name, timeout);
```

Comment:
I'm going to suggest that you put a comment above this with a link to the Connext documentation, and a small note saying that we are intentionally getting a copy of the `DDS::Topic` for each thread.  Then it is a little clearer in the future why we are always freeing during destruction.

Separate nit: Since the timeout we are going to use is fixed, you could combine this line and the previous one:

```
DDS::Topic * topic = participant->find_topic(dds_topic_name, DDS::Duration_t::from_seconds(0));
```

But I'll leave it to you what you find more readable.

Reply:
Done in https://github.com/ros2/rmw_connext/pull/444/commits/7a23fd09fb76bd7a01682a2100f23c2971ad20f1

Reply:
I forgot adding the comment, see https://github.com/ros2/rmw_connext/pull/444/commits/3c92dcfc0d6069d496d9d34bf76bc77831169d19.